cmake/FindSphinx.cmake

```
@@ -29,20 +17,19 @@ if(Sphinx_FIND_VERSION)
     set(QUERY_VERSION "${QUERY_VERSION_ERR}")
   endif()
 
-  if(NOT QUERY_VERSION_RESULT)
-    string(REGEX MATCH "[0-9.]+" SPHINX_VERSION "${QUERY_VERSION}")
-  endif()
+  string(REGEX MATCH "[0-9.]+" SPHINX_VERSION "${QUERY_VERSION}")
+endif()
 
-  set(SPHINX_VERSION_COMPATIBLE TRUE)
-  # Blacklist broken version
-  if("${SPHINX_VERSION}" VERSION_EQUAL "2.1.0")
-    message(WARNING "Sphinx version 2.1.0 is not compatible.")
-    set(SPHINX_VERSION_COMPATIBLE FALSE)
-  endif()
+set(SPHINX_VERSION_COMPATIBLE TRUE)
+# Blacklist broken version
+if("${SPHINX_VERSION}" VERSION_EQUAL "2.1.0")
+  message(WARNING "Sphinx version 2.1.0 is not compatible.")
```

to avoid conflicts with the Sphinx PR in #3642, we could either integrate this commit in #3642, or apply this patch:
```suggestion
if("${SPHINX_VERSION}" VERSION_EQUAL "2.1.0" OR "${SPHINX_VERSION}" VERSION_EQUAL "3.0.0")
  message(WARNING "Sphinx version ${SPHINX_VERSION} is not compatible.")
```

---

that is a trivial merge, right?

---

yes, but it's an annoying backport in 4.1.3 because merge commits break `git cherry-pick`